NoDenotation.owner assertion error

## Compiler version

3.0.0-RC1

## Minimized code

```Scala
import scala.deriving._

trait TupleConversion[A, B] {
  def to(a: A): B
  def from(b: B): A
}

object TupleConversion  {
  inline given autoTupleConversion[Prod <: Product](using m: Mirror.ProductOf[Prod]): TupleConversion[Prod, m.MirroredElemTypes] =
    new TupleConversion[Prod, m.MirroredElemTypes] {
      def to(a: Prod): m.MirroredElemTypes = Tuple.fromProductTyped(a)
      def from(b: m.MirroredElemTypes): Prod = m.fromProduct(b)
    }
}

final case class Data(s0: Int, s1: Int)

abstract class BaseSpec(f: () => Unit)

object ProductBuilderTest
  extends BaseSpec(() => {
    val conv = implicitly[TupleConversion[Data, (Int, Int)]]
  })
```

Note that the error only happens if the implicit search is happening from the function passed to the base constructor.

## Output (click arrow to expand)
<details>

```scala
java.lang.AssertionError: NoDenotation.owner while compiling demo.scala
[error] ## Exception when compiling 1 sources to target\scala-3.0.0-RC1\classes
[error] java.lang.AssertionError: NoDenotation.owner
[error] dotty.tools.dotc.core.SymDenotations$NoDenotation$.owner(SymDenotations.scala:2394)
[error] dotty.tools.dotc.transform.HoistSuperArgs$Hoister.dotty$tools$dotc$transform$HoistSuperArgs$Hoister$$_$needsRewire$1(HoistSuperArgs.scala:122)
[error] dotty.tools.dotc.transform.HoistSuperArgs$$anon$1.apply(HoistSuperArgs.scala:139)
[error] dotty.tools.dotc.core.Types$TypeMap.op$proxy13$1(Types.scala:5130)
[error] dotty.tools.dotc.core.Types$TypeMap.mapArgs(Types.scala:5130)
[error] dotty.tools.dotc.core.Types$TypeMap.mapArgs(Types.scala:5131)
[error] dotty.tools.dotc.core.Types$TypeMap.mapOver(Types.scala:5165)
[error] dotty.tools.dotc.transform.HoistSuperArgs$$anon$1.apply(HoistSuperArgs.scala:145)
[error] dotty.tools.dotc.transform.HoistSuperArgs$$anon$1.apply(HoistSuperArgs.scala:138)
[error] dotty.tools.dotc.ast.TreeTypeMap.mapType(TreeTypeMap.scala:63)
[error] dotty.tools.dotc.ast.TreeTypeMap.transform(TreeTypeMap.scala:90)
[error] dotty.tools.dotc.ast.TreeTypeMap.transform(TreeTypeMap.scala:111)
[error] dotty.tools.dotc.ast.Trees$Instance$TreeMap.transform$$anonfun$1(Trees.scala:1422)
[error] scala.collection.immutable.List.mapConserve(List.scala:472)
[error] dotty.tools.dotc.ast.Trees$Instance$TreeMap.transform(Trees.scala:1422)
[error] dotty.tools.dotc.ast.Trees$Instance$TreeMap.transform(Trees.scala:1321)
[error] dotty.tools.dotc.ast.TreeTypeMap.transform(TreeTypeMap.scala:129)
[error] dotty.tools.dotc.ast.Trees$Instance$TreeMap.transform(Trees.scala:1389)
[error] dotty.tools.dotc.ast.TreeTypeMap.transform(TreeTypeMap.scala:129)
[error] dotty.tools.dotc.ast.Trees$Instance$TreeMap.transform$$anonfun$1(Trees.scala:1422)
[error] scala.collection.immutable.List.mapConserve(List.scala:472)
[error] dotty.tools.dotc.ast.Trees$Instance$TreeMap.transform(Trees.scala:1422)
[error] dotty.tools.dotc.ast.Trees$Instance$TreeMap.transformSub(Trees.scala:1426)
[error] dotty.tools.dotc.ast.TreeTypeMap.transformDefs(TreeTypeMap.scala:138)
[error] dotty.tools.dotc.ast.TreeTypeMap.transform(TreeTypeMap.scala:106)
[error] dotty.tools.dotc.ast.TreeTypeMap.transform(TreeTypeMap.scala:95)
[error] dotty.tools.dotc.ast.Trees$Instance$TreeMap.transform$$anonfun$1(Trees.scala:1422)
[error] scala.collection.immutable.List.mapConserve(List.scala:472)
[error] dotty.tools.dotc.ast.Trees$Instance$TreeMap.transform(Trees.scala:1422)
[error] dotty.tools.dotc.ast.Trees$Instance$TreeMap.transformSub(Trees.scala:1426)
[error] dotty.tools.dotc.ast.TreeTypeMap.transformDefs(TreeTypeMap.scala:138)
[error] dotty.tools.dotc.ast.TreeTypeMap.transform(TreeTypeMap.scala:106)
[error] dotty.tools.dotc.ast.TreeTypeMap.apply(TreeTypeMap.scala:151)
[error] dotty.tools.dotc.transform.HoistSuperArgs$Hoister.$anonfun$1(HoistSuperArgs.scala:154)
[error] dotty.tools.dotc.ast.tpd$.DefDef(tpd.scala:283)
[error] dotty.tools.dotc.transform.HoistSuperArgs$Hoister.hoistSuperArg(HoistSuperArgs.scala:155)
[error] dotty.tools.dotc.transform.HoistSuperArgs$Hoister.hoistSuperArgsFromCall$$anonfun$1(HoistSuperArgs.scala:179)
[error] dotty.tools.dotc.core.Decorators$ListDecorator$.loop$1(Decorators.scala:92)
[error] dotty.tools.dotc.core.Decorators$ListDecorator$.mapconserve$extension(Decorators.scala:108)
[error] dotty.tools.dotc.transform.HoistSuperArgs$Hoister.hoistSuperArgsFromCall(HoistSuperArgs.scala:179)
[error] dotty.tools.dotc.transform.HoistSuperArgs.transformTypeDef(HoistSuperArgs.scala:205)
[error] dotty.tools.dotc.transform.MegaPhase.goTypeDef(MegaPhase.scala:989)
[error] dotty.tools.dotc.transform.MegaPhase.transformNamed$1(MegaPhase.scala:257)
[error] dotty.tools.dotc.transform.MegaPhase.transformTree(MegaPhase.scala:427)
[error] dotty.tools.dotc.transform.MegaPhase.transformStat$2(MegaPhase.scala:437)
[error] dotty.tools.dotc.transform.MegaPhase.recur$1(MegaPhase.scala:442)
[error] dotty.tools.dotc.transform.MegaPhase.recur$1(MegaPhase.scala:442)
[error] dotty.tools.dotc.transform.MegaPhase.recur$1(MegaPhase.scala:442)
[error] dotty.tools.dotc.transform.MegaPhase.recur$1(MegaPhase.scala:442)
[error] dotty.tools.dotc.transform.MegaPhase.recur$1(MegaPhase.scala:442)
[error] dotty.tools.dotc.transform.MegaPhase.recur$1(MegaPhase.scala:442)
[error] dotty.tools.dotc.transform.MegaPhase.recur$1(MegaPhase.scala:442)
[error] dotty.tools.dotc.transform.MegaPhase.recur$1(MegaPhase.scala:442)
[error] dotty.tools.dotc.transform.MegaPhase.recur$1(MegaPhase.scala:442)
[error] dotty.tools.dotc.transform.MegaPhase.transformStats(MegaPhase.scala:442)
[error] dotty.tools.dotc.transform.MegaPhase.mapPackage$1(MegaPhase.scala:382)
[error] dotty.tools.dotc.transform.MegaPhase.transformUnnamed$1(MegaPhase.scala:385)
[error] dotty.tools.dotc.transform.MegaPhase.transformTree(MegaPhase.scala:429)
[error] dotty.tools.dotc.transform.MegaPhase.transformUnit(MegaPhase.scala:448)
[error] dotty.tools.dotc.transform.MegaPhase.run(MegaPhase.scala:460)
[error] dotty.tools.dotc.core.Phases$Phase.runOn$$anonfun$1(Phases.scala:303)
[error] scala.collection.immutable.List.map(List.scala:246)
[error] dotty.tools.dotc.core.Phases$Phase.runOn(Phases.scala:304)
[error] dotty.tools.dotc.Run.runPhases$4$$anonfun$4(Run.scala:205)
[error] scala.runtime.function.JProcedure1.apply(JProcedure1.java:15)
[error] scala.runtime.function.JProcedure1.apply(JProcedure1.java:10)
[error] scala.collection.ArrayOps$.foreach$extension(ArrayOps.scala:1323)
[error] dotty.tools.dotc.Run.runPhases$5(Run.scala:215)
[error] dotty.tools.dotc.Run.compileUnits$$anonfun$1(Run.scala:223)
[error] scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.scala:18)
[error] dotty.tools.dotc.util.Stats$.maybeMonitored(Stats.scala:67)
[error] dotty.tools.dotc.Run.compileUnits(Run.scala:230)
[error] dotty.tools.dotc.Run.compileSources(Run.scala:166)
[error] dotty.tools.dotc.Run.compile(Run.scala:150)
[error] dotty.tools.dotc.Driver.doCompile(Driver.scala:39)
[error] dotty.tools.xsbt.CompilerBridgeDriver.run(CompilerBridgeDriver.java:88)
[error] dotty.tools.xsbt.CompilerBridge.run(CompilerBridge.java:22)
[error] sbt.internal.inc.AnalyzingCompiler.compile(AnalyzingCompiler.scala:92)
[error] sbt.internal.inc.MixedAnalyzingCompiler.$anonfun$compile$7(MixedAnalyzingCompiler.scala:186)
[error] scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:23)
[error] sbt.internal.inc.MixedAnalyzingCompiler.timed(MixedAnalyzingCompiler.scala:241)
[error] sbt.internal.inc.MixedAnalyzingCompiler.$anonfun$compile$4(MixedAnalyzingCompiler.scala:176)
[error] sbt.internal.inc.MixedAnalyzingCompiler.$anonfun$compile$4$adapted(MixedAnalyzingCompiler.scala:157)
[error] sbt.internal.inc.JarUtils$.withPreviousJar(JarUtils.scala:239)
[error] sbt.internal.inc.MixedAnalyzingCompiler.compileScala$1(MixedAnalyzingCompiler.scala:157)
[error] sbt.internal.inc.MixedAnalyzingCompiler.compile(MixedAnalyzingCompiler.scala:204)
[error] sbt.internal.inc.IncrementalCompilerImpl.$anonfun$compileInternal$1(IncrementalCompilerImpl.scala:573)
[error] sbt.internal.inc.IncrementalCompilerImpl.$anonfun$compileInternal$1$adapted(IncrementalCompilerImpl.scala:573)
[error] sbt.internal.inc.Incremental$.$anonfun$apply$5(Incremental.scala:174)
[error] sbt.internal.inc.Incremental$.$anonfun$apply$5$adapted(Incremental.scala:172)
[error] sbt.internal.inc.Incremental$$anon$2.run(Incremental.scala:459)
[error] sbt.internal.inc.IncrementalCommon$CycleState.next(IncrementalCommon.scala:116)
[error] sbt.internal.inc.IncrementalCommon$$anon$1.next(IncrementalCommon.scala:56)
[error] sbt.internal.inc.IncrementalCommon$$anon$1.next(IncrementalCommon.scala:52)
[error] sbt.internal.inc.IncrementalCommon.cycle(IncrementalCommon.scala:261)
[error] sbt.internal.inc.Incremental$.$anonfun$incrementalCompile$8(Incremental.scala:414)
[error] sbt.internal.inc.Incremental$.withClassfileManager(Incremental.scala:499)
[error] sbt.internal.inc.Incremental$.incrementalCompile(Incremental.scala:401)
[error] sbt.internal.inc.Incremental$.apply(Incremental.scala:166)
[error] sbt.internal.inc.IncrementalCompilerImpl.compileInternal(IncrementalCompilerImpl.scala:573)
[error] sbt.internal.inc.IncrementalCompilerImpl.$anonfun$compileIncrementally$1(IncrementalCompilerImpl.scala:491)
[error] sbt.internal.inc.IncrementalCompilerImpl.handleCompilationError(IncrementalCompilerImpl.scala:332)
[error] sbt.internal.inc.IncrementalCompilerImpl.compileIncrementally(IncrementalCompilerImpl.scala:420)
[error] sbt.internal.inc.IncrementalCompilerImpl.compile(IncrementalCompilerImpl.scala:137)
[error] sbt.Defaults$.compileIncrementalTaskImpl(Defaults.scala:2177)
[error] sbt.Defaults$.$anonfun$compileIncrementalTask$2(Defaults.scala:2134)
[error] sbt.internal.io.Retry$.apply(Retry.scala:40)
[error] sbt.internal.io.Retry$.apply(Retry.scala:23)
[error] sbt.internal.server.BspCompileTask$.compute(BspCompileTask.scala:31)
[error] sbt.Defaults$.$anonfun$compileIncrementalTask$1(Defaults.scala:2130)
[error] scala.Function1.$anonfun$compose$1(Function1.scala:49)
[error] sbt.internal.util.$tilde$greater.$anonfun$$u2219$1(TypeFunctions.scala:62)
[error] sbt.std.Transform$$anon$4.work(Transform.scala:68)
[error] sbt.Execute.$anonfun$submit$2(Execute.scala:282)
[error] sbt.internal.util.ErrorHandling$.wideConvert(ErrorHandling.scala:23)
[error] sbt.Execute.work(Execute.scala:291)
[error] sbt.Execute.$anonfun$submit$1(Execute.scala:282)
[error] sbt.ConcurrentRestrictions$$anon$4.$anonfun$submitValid$1(ConcurrentRestrictions.scala:265)
[error] sbt.CompletionService$$anon$2.call(CompletionService.scala:64)
[error] java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
[error] java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
[error] java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
[error] java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1130)
[error] java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:630)
[error] java.base/java.lang.Thread.run(Thread.java:832)
```
</details>

--- a/compiler/src/dotty/tools/dotc/transform/HoistSuperArgs.scala
+++ b/compiler/src/dotty/tools/dotc/transform/HoistSuperArgs.scala
@@ -119,7 +119,8 @@ class HoistSuperArgs extends MiniPhase with IdentityDenotTransformer { thisPhase
       /** Only rewire types that are owned by the current Hoister and is an param or accessor */
       def needsRewire(tp: Type) = tp match {
         case ntp: NamedType =>
-          (ntp.symbol.owner == cls || ntp.symbol.owner == constr) && ntp.symbol.isParamOrAccessor
+          val owner = ntp.symbol.maybeOwner
+          (owner == cls || owner == constr) && ntp.symbol.isParamOrAccessor
         case _ => false
       }
 

--- a/tests/pos/i11732.scala
+++ b/tests/pos/i11732.scala
@@ -1,0 +1,23 @@
+import scala.deriving._
+
+trait TupleConversion[A, B] {
+  def to(a: A): B
+  def from(b: B): A
+}
+
+object TupleConversion  {
+  inline given autoTupleConversion[Prod <: Product](using m: Mirror.ProductOf[Prod]): TupleConversion[Prod, m.MirroredElemTypes] =
+    new TupleConversion[Prod, m.MirroredElemTypes] {
+      def to(a: Prod): m.MirroredElemTypes = Tuple.fromProductTyped(a)
+      def from(b: m.MirroredElemTypes): Prod = m.fromProduct(b)
+    }
+}
+
+final case class Data(s0: Int, s1: Int)
+
+abstract class BaseSpec(f: () => Unit)
+
+object ProductBuilderTest
+  extends BaseSpec(() => {
+    val conv = implicitly[TupleConversion[Data, (Int, Int)]]
+  })

